src/components/Synthetics/PositionSeller/PositionSeller.tsx

```diff
@@ -325,8 +368,29 @@ export function PositionSeller(p: Props) {
         setPendingOrder,
         setPendingTxns,
         setPendingPosition,
-      }
-    );
+      },
+      metricId
+    )
+      .then(() => {
+        metrics.sendMetric({
+          event: `${metricType}.sent`,
+          isError: false,
+          fields: metrics.getPendingEvent(metricId),
+        });
+        metrics.startTimer(metricId);
+
+        return Promise.resolve();
+      })
+      .catch((error) => {
+        metrics.sendMetric({
+          event: `${metricType}.${isUserRejectedActionError(error) ? "rejected" : "fail"}`,
+          isError: true,
+          message: error.message,
+          fields: metrics.getPendingEvent(metricId, true),
+        });
+
+        throw error;
+      });
```

Comment:
this logic seems duplicated several times, let's create some factory/promise-wrapper function that would do same thing for all such places